src/library/scala/collection/immutable/List.scala

```diff
@@ -470,6 +475,7 @@ sealed abstract class List[+A]
 
 case class :: [+A](override val head: A, private[scala] var next: List[A @uncheckedVariance]) // sound because `next` is used only locally
```

Comment:
Is :: meant to be extendable?

Reply:
It could be final. `List` is sealed, but that doesn't prevent non-Scala subclasses.

Reply:
It doesn't prevent *Scala* subclasses either!
```scala
$ scala
Welcome to Scala 2.13.0-M4 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_40).
Type in expressions for evaluation. Or try :help.

scala> class Oops extends ::(5, Nil)
defined class Oops
```

Reply:
I _still_ forget about that!

Reply:
*final* all the things!